classes/image_type_torizon.bbclass

```diff
@@ -14,7 +14,11 @@ python() {
         d.setVar('TEZI_CONFIG_FORMAT', torizon_tezi)
 }
 
-TEZI_ROOT_FSOPTS:append:torizon-signed = " -O verity"
+# TODO: Consider always enabling verity.
+#       This would facilitate upgrades from "tdx-signed" to "torzion-signed";
+#       If not done beforehand, the upgrade process would require running
+#       "tune2fs -O verity" on the device.
```

Comment:
Question about this comment, as it is now, what would happen in such a theoretical update? I imagine it would just fail to boot into the new image and rollback? Or is it something more dramatic than that?

Reply:
In this case I think the process would be more or less like this:

1) First, upgrade to a version having "optional" support for composefs+verity on the ramdisk.
2) Second, do another update enabling verity; I think the update would fail at this point when pulling the update because libostree wouldn't be able enable verity on the downloaded files; but I'm not totally sure. If it fails, the error is reported to the platform and that's it. Otherwise, the system would finish the download, reboot and probably fail when mounting the root mount which is done by `ostree-prepare-root`. Today, if that happens the system would simply hang in the last step performed by the ramdisk.

Anyways, providing a reliable path for upgrading, which would include allowing rollbacks, is a "to-be-created" ticket in my personal backlog.

Reply:
Okay then that's good to know. This fsverity stuff introduces a lot of nuances and "gotchas" with regards to update paths and such.